include/boost/utility/string_ref.hpp

```diff
@@ -26,7 +27,6 @@
 #include <algorithm>
 #include <iterator>
 #include <string>
-#include <iosfwd>
```

Comment:
`iosfwd` is still needed for `basic_ostream` in `operator<<` declaration.

Reply:
It is still being included (from `ostream_write.hpp` which is included here).

Reply:
I don't think relying on transitive includes is correct. My rule of thumb is include everything you directly use in the header.

Reply:
No objection to employing that rule here. (Done).